src/dimension_slice.c

```diff
@@ -171,12 +170,11 @@ dimension_vec_tuple_found_list(TupleInfo *ti, void *data)
 			break;
 	}
 
-	old = MemoryContextSwitchTo(ti->mctx);
-	slice = dimension_slice_from_slot(ti->slot);
-	Assert(NULL != slice);
-	*slices = lappend(*slices, slice);
-	MemoryContextSwitchTo(old);
-
+	TS_WITH_MEMORY_CONTEXT(ti->mctx, {
+		slice = dimension_slice_from_slot(ti->slot);
```

Comment:
I think you can also declare `slice` in this line. 

```suggestion
		DimensionSlice *slice = dimension_slice_from_slot(ti->slot);
```